File: LeetCode/java/1817_Finding_the_Users_Active_Minutes.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

class Solution {
    public int[] findingUsersActiveMinutes(int[][] logs, int k) {
        int[] answer = new int[k];

        Map<Integer, Set<Integer>> map = new HashMap<>();

        for (int[] log : logs) {
            if (!map.containsKey(log[0])) {
                map.put(log[0], new HashSet<Integer>());
            }

            map.get(log[0]).add(log[1]);
        }

        Map<Integer, Integer> record = new HashMap<>();

        for (int key : map.keySet()) {
            answer[map.get(key).size() - 1]++;
        }

        return answer;
    }
}

File: LeetCode/cpp/1817_Finding_the_Users_Active_Minutes.cpp
class Solution {
public:
	vector<int> findingUsersActiveMinutes(vector<vector<int>>& logs, int k) {
		vector<int> answer(k, 0);

		unordered_map<int, set<int>> um;

		for (vector<int> log : logs) {
			um[log[0]].insert(log[1]);
		}

		unordered_map<int, int> record;

		for (auto m : um) {
			answer[m.second.size() - 1]++;
		}

		return answer;
	}
};